src/Tools/NodeDocumentationMarkdownGenerator/AssemblyHandler.cs

```diff
@@ -0,0 +1,387 @@
+            {
+                return new List<FunctionDescriptor>();
+            }
+
+            // Getting ZT imports from CLRModuleTypes
+            DLLModule dllModule = null;
+            dllModule = DLLFFIHandler.GetModuleForInspection(asm);
+
+            dllModule.ScanModule(true);
+            List<CLRModuleType> moduleTypes = CLRModuleType.GetTypes((CLRModuleType mtype) => { return mtype.Module == dllModule; });
+
+            var customizationFile = LibraryCustomizationServices.GetForAssembly(asm.Location, pathManager);
+
+            foreach (var t in moduleTypes)
+            {
+                try
+                {
+                    var className = t.ClassNode.ClassName;
+                    var externalLib = t.ClassNode.ExternLibName;
+
+                    // For some reason mscorelib sometimes gets pass to here, so filtering it away.
+                    if (t.CLRType.Assembly.GetName().Name == objectAsmName ||
+                        t.ClassNode.ClassAttributes.HiddenInLibrary) continue;
+
+                    var ctorNodes = t.ClassNode.Procedures
+                        .OfType<ConstructorDefinitionNode>()
+                        .Where(c => c.Access == ProtoCore.CompilerDefinitions.AccessModifier.Public &&
+                                    c.MethodAttributes != null ?
+                                        !c.MethodAttributes.IsObsolete && !c.MethodAttributes.HiddenInLibrary :
+                                        true)
+                        .Cast<AssociativeNode>();
+
+                    var functionNodes = t.ClassNode.Procedures
+                        .OfType<FunctionDefinitionNode>()
+                        .Where(f => !f.MethodAttributes.HiddenInLibrary &&
+                                    !f.MethodAttributes.IsObsolete &&
+                                    f.Name != "_Dispose")
+                        .Cast<AssociativeNode>();
+
+                    var associativeNodes = ctorNodes.Union(functionNodes);
+
+                    foreach (var node in associativeNodes)
+                    {
+                        if (TryGetFunctionDescriptor(node, asm, asm.Location, className, customizationFile, out ReflectionFunctionDescriptor des))
+                        {
+                            descriptors.Add(des);
+                        }
+                    }
+                }
+                catch (Exception)
+                {
+                    continue;
+                }
+            }
+
+            return descriptors;
+        }
+
+        private static List<FunctionDescriptor> GetFunctionDescriptorsFromDs(PathManager pathManager, string dsFilePath)
+        {
+            var descriptors = new List<FunctionDescriptor>();
+
+            var t = new ImportModuleHandler(new ProtoCore.Core(new ProtoCore.Options()));
+            var dsCodeNode = t.Import(dsFilePath, "", "");
+            var classNode = dsCodeNode.CodeNode.Body.OfType<ClassDeclNode>().FirstOrDefault();
```

Comment:
couldn't it be possible theres more one than one class defined in a .ds file?

Reply:
i guess, im not that familiar with ds files..

However, this seems to be associated with more issues than i thought. This is giving me some failures now since i moved this solution to its own bin folder instead of dynamos main. I think the issue is that ImportModuleHandler.Import() uses nonreflection methods and therefore it fails.

If you had ideas about avoiding reflection it would probably be good to chat on that.